docs/sharded-computation.md

```diff
@@ -14,15 +14,21 @@ kernelspec:
 (sharded-computation)=
 # Introduction to sharded computation
 
-JAX's {class}`jax.Array` object is designed with distributed data and computation in mind.
+This tutorial serves as an introduction to device parallelism for Single-Program Multi-Data (SPMD) code in JAX. SPMD is a parallelism technique where the same computation, such as the forward pass of a neural network, can be run on different input data (for example, different inputs in a batch) in parallel on different devices, such as several GPUs or Google TPUs.
 
-This section will cover three modes of parallel computation:
+The tutorial covers three modes of parallel computation:
 
-- Automatic parallelism via {func}`jax.jit`, in which we let the compiler choose the optimal computation strategy
-- Semi-automatic parallelism using {func}`jax.jit` and {func}`jax.lax.with_sharding_constraint`
-- Fully manual parallelism using {func}`jax.experimental.shard_map.shard_map`
+- _Automatic parallelism via {func}`jax.jit`_: The compiler chooses the optimal computation strategy (a.k.a. "the compiler takes the wheel").
+- _Semi-automated parallelism_ using {func}`jax.jit` and {func}`jax.lax.with_sharding_constraint`
+- _Fully manual parallelism with manual control using {func}`jax.experimental.shard_map.shard_map`_: `shard_map` enables per-device code and explicit communication collectives
 
-These examples will be run on Colab's free TPU runtime, which provides eight devices to work with:
+**Note:** With {func}`jax.jit`, based on the sharding of the data, the compiler automatically partitions bulk array functions over multiple devices, dividing shardings for intermediate and output values, and parallelizes their evaluation, even inserting communication operations when necessary.
+
+**Note:** It's possible to use `shard_map` inside {func}`jax.jit`, for example, to enable manual collectives and hand-optimization when needed.
```

Comment:
These notes seem out-of-place here. Delete?

Reply:
OK!